internal/commands/pull/inputs.go

```diff
@@ -109,3 +117,62 @@ func (i *inputs) resolveRemoteApp(ui terminal.UI, clients cli.Clients) (realm.Ap
 
 	return app, nil
 }
+
+func (i *inputs) resolveClient(ui terminal.UI, realmClient realm.Client, groupID, appID string) (map[string]*zip.Reader, error) {
+	compatibleTemplates, ok, err := realmClient.CompatibleTemplates(groupID, appID)
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		return nil, nil
+	}
+
+	if i.TemplateID != "" {
+		for _, template := range compatibleTemplates {
+			// This is the ID that they selected
+			if template.ID == i.TemplateID {
+				// Fetch the template
+				templateZip, err := realmClient.ClientTemplate(groupID, appID, template.ID)
+				if err != nil {
+					return nil, err
+				}
+				return map[string]*zip.Reader{template.ID: templateZip}, nil
+			}
+		}
+		return nil, fmt.Errorf("template %s is not compatible with this app", i.TemplateID)
```

Comment:
should probably wrap this with `'%s'`

Reply:
`%q` works as well!

Reply:
sort of an arbitrary distinction, but go with single quotes as that what most of our error messages are currently formatted with (when needed).  we use the double-quotes everywhere in help text.  i imagine at some point it will be normalized, but best to keep the different variants consistent within themselves at first